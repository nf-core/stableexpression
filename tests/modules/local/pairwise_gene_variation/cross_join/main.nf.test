nextflow_process {

    name "Test Process CROSS_JOIN"
    script "modules/local/pairwise_gene_variation/cross_join/main.nf"
    process "CROSS_JOIN"
    tag "cross_join"
    tag "module"

    test("Should run without failures") {

        when {
            process {
                """
                meta = [index_1: 0, index_2: 1]
                file_1 = file( '$projectDir/tests/test_data/pairwise_gene_variation/make_chunks/output/count_chunk.0.parquet', checkIfExists: true)
                file_2 = file( '$projectDir/tests/test_data/pairwise_gene_variation/make_chunks/output/count_chunk.1.parquet', checkIfExists: true)
                input[0] = [meta, file_1, file_2]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
