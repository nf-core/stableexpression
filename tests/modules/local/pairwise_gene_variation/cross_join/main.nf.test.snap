{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "cross_join.0.1.parquet:md5,ef1d4777f59bf8c52f05fa37d638989f"
                ],
                "1": [
                    [
                        "CROSS_JOIN",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "CROSS_JOIN",
                        "polars",
                        "1.17.1"
                    ]
                ],
                "data": [
                    "cross_join.0.1.parquet:md5,ef1d4777f59bf8c52f05fa37d638989f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-29T15:29:44.996421136"
    }
}